### ISSUE TITLE
Support file upload without special keys

### ISSUE THREAD

**Issue by vertti** [tiangolo/fastapi#1653]
### Example

Your default example for uploading files:
```Python
from fastapi import FastAPI, UploadFile

app = FastAPI()

@app.post("/file/")
async def create_upload_file(file: UploadFile = File(...)):
    return {"filename": file.filename}
```

You can't use any default method of uploading with this endpoint as it does not support the whole body being the uploded file. Instead it expects to find a key-value-pair that matches the UploadFile parameter name (`file` in the above example).

So if you for example use Postman to create your POST request and select [x] binary body and a file to upload, or if you use curl with:

```
curl -X POST --data-binary @example.txt http://localhost/file/
```
FastApi will fail with:
```
{"detail":[{"loc":["body","file"],"msg":"field required","type":"value_error.missing"}]}
```
as it expects to find `file`.

Yes, I know I can add that key but I think the default should be default of your most used tools. And it's cumbersome to explain such special requirements for the users of your API.

### Environment

* OS: Windows
* FastAPI Version: 0.58.1
* Python version: 3.8

**Comment by phy25** [comment]
The doc mentioned that `File` support focuses on form data content type. If you send it without form data, you can try the code here: https://github.com/tiangolo/fastapi/issues/579#issuecomment-536797166

**Comment by tiangolo** [comment]
Thanks for the help here @phy25 ! :coffee: 

@vertti the default of the most used tools depends on which tools you use the most and which defaults you use. The current default behavior makes it compatible with web browsers, tools like `requests` and `httpx`, etc. I also think it's a bit weird that `multipart/form-data` is one of the most common ways to upload files. At the same time, it's also the main way to upload several files at the same time.

But you can always use the request directly as @phy25 mentions.

If you need to use `curl`, then you can also use:

```console
$ curl http://localhost:8008/file/ -F file=@example.txt
```

If that solves your use case, you can close the issue. :heavy_check_mark:

**Comment by vertti** [comment]
Yeah, this got sorted and I appreciate the help and the generally very good documentation of the FastAPI. Thanks!
